Python/bytecodes.c

```diff
@@ -3015,10 +3015,20 @@ dummy_func(
 
 
         op(_ITER_CHECK_LIST, (iter -- iter)) {
-            EXIT_IF(Py_TYPE(PyStackRef_AsPyObjectBorrow(iter)) != &PyListIter_Type);
+            PyObject *iter_o = PyStackRef_AsPyObjectBorrow(iter);
+            EXIT_IF(Py_TYPE(iter_o) != &PyListIter_Type);
+#ifdef Py_GIL_DISABLED
+            _PyListIterObject *it = (_PyListIterObject *)iter_o;
+            EXIT_IF(it->it_seq == NULL ||
+                    !_Py_IsOwnedByCurrentThread((PyObject *)it->it_seq) ||
+                    !_PyObject_GC_IS_SHARED(it->it_seq));
+#endif
         }
 
         replaced op(_ITER_JUMP_LIST, (iter -- iter)) {
+// For free-threaded Python, the loop exit can happen at any point during item
+// retrieval, so separate ops don't make much sense.
+#ifndef Py_GIL_DISABLED
```

Comment:
Remove the nested `#ifndef Py_GIL_DISABLED` below (line 3039) if this whole uop is guarded by an `ifndef Py_GIL_DISABLED`.